storage/client/queries/queries.go

```diff
@@ -24,13 +24,39 @@ const (
 			WHERE layer=$1`
 
 	Blocks = `
-		SELECT height, block_hash, time, num_txs
-			FROM chain.blocks
-			WHERE ($1::bigint IS NULL OR height >= $1::bigint) AND
-						($2::bigint IS NULL OR height <= $2::bigint) AND
-						($3::timestamptz IS NULL OR time >= $3::timestamptz) AND
-						($4::timestamptz IS NULL OR time < $4::timestamptz) AND
-						($5::text IS NULL OR block_hash = $5::text)
+		SELECT
+			height,
+			block_hash,
+			time,
+			num_txs,
+			metadata,
+			ROW(
+			    proposer_node.id,
+			    proposer_node.entity_id,
+			    proposer_entity.address,
+			    proposer_entity.meta
+			) AS proposer_node_info,
+			ARRAY(
```

Comment:
:hot_face: swanky. nice job.

Maybe a comment for future readers? `-- Maps signer_node_consensus_pubkey_addresses to an array of structs that each describe the corresponding node` or sth